rotors_gazebo_plugins/src/gazebo_wind_plugin.cpp

```diff
@@ -119,44 +135,151 @@ void GazeboWindPlugin::OnUpdate(const common::UpdateInfo& _info) {
+        (top_z_[idx_i[2 * i] + idx_j[2 * i] * n_x_] - bottom_z_[idx_i[2 * i] +
+                                                            idx_j[2 * i] * n_x_]);
+    }
+
+    float vertical_factors_min = std::min(std::min(std::min(
+      vertical_factors_columns[0], vertical_factors_columns[1]),
+      vertical_factors_columns[2]), vertical_factors_columns[3]);
+    float vertical_factors_max = std::max(std::max(std::max(
+      vertical_factors_columns[0], vertical_factors_columns[1]),
+      vertical_factors_columns[2]), vertical_factors_columns[3]);
+
+    // Check if aircraft is out of wind field or not, and act accordingly.
+    if (!(i_inf < 0 || j_inf < 0 || vertical_factors_max < 0 || 
+        i_sup > (n_x_ - 1) || j_sup > (n_y_ - 1) || vertical_factors_min > 1)) {
+      // Find indices in z-direction for each of the vertices. If link is not 
+      // within the range of one of the columns, set to lowest or highest two.
+      int idx_k[8] = {0, static_cast<int>(vertical_spacing_factors_.size()) - 1, 0,
+                      static_cast<int>(vertical_spacing_factors_.size()) - 1, 0,
+                      static_cast<int>(vertical_spacing_factors_.size()) - 1, 0,
+                      static_cast<int>(vertical_spacing_factors_.size()) - 1};
+
+      for (int i = 0; i < 4; i++) {
+        if (vertical_factors_columns[i] < 0) {
+          // Link z-position below lowest grid point of that column.
+          idx_k[2 * i + 1] = 1;
+        } else if (vertical_factors_columns[i] >= 1) {
+          // Link z-position above highest grid point of that column.
+          idx_k[2 * i] = vertical_spacing_factors_.size() - 2;
+        } else {
+          // Link z-position between two grid points in that column.
+          for (int j = 0; j < vertical_spacing_factors_.size() - 1; j++) {
+            if (vertical_spacing_factors_[j] <= vertical_factors_columns[i] && 
+                vertical_spacing_factors_[j + 1] > vertical_factors_columns[i]) {
+              idx_k[2 * i] = j;
+              idx_k[2 * i + 1] = j + 1;
+              break;
+            }
+          }
+        }
+      }
+
+      // Extract the wind velocities corresponding to each vertex.
+      math::Vector3 wind_at_vertices[8];
+      for (int i = 0; i < 8; i++) {
+        wind_at_vertices[i].x = u_[idx_i[i] + idx_j[i] * n_x_ + idx_k[i] * n_x_
+                                                                        * n_y_];
+        wind_at_vertices[i].y = v_[idx_i[i] + idx_j[i] * n_x_ + idx_k[i] * n_x_
+                                                                        * n_y_];
+        wind_at_vertices[i].z = w_[idx_i[i] + idx_j[i] * n_x_ + idx_k[i] * n_x_
+                                                                        * n_y_];
+      }
+
+      // Extract the relevant coordinate of every point needed for trilinear 
+      // interpolation (first z-direction, then x-direction, then y-direction).
+      float interpolation_points[14];
```

Comment:
I was using floats for all variables read from the wind field text file and all others calculated with these to save disk space and computational time.
Do you want me to change them to doubles to be consistent with the other variables used in the interpolation function (gazebo::math::Vector3)?